src/platform/windows/window.rs

```diff
@@ -715,15 +715,43 @@ unsafe fn init(
         events_loop_proxy,
     };
 
-    if let Some(_) = attributes.fullscreen {
-        win.set_fullscreen(attributes.fullscreen);
+    if let Some(_) = &attributes.fullscreen {
+        win.set_fullscreen(attributes.fullscreen.clone());
         force_window_active(win.window.0);
     }
 
     if let Some(dimensions) = attributes.dimensions {
         win.set_inner_size(dimensions);
     }
 
+    if attributes.center_window && !&attributes.fullscreen.is_some(){
+        let window_size = win.get_outer_size();
+
+        let monitor;
+        if attributes.start_monitor == -1 {
```

Comment:
Why is `start_monitor` an `i16`? This should be an `Option<MonitorId>` instead.

Reply:
This is done for simplicity. only supplying a monitors index in display config is better IMO, start_monitor is renamed to monitor_index to better reflect that.

Reply:
We don't really have any guarantee that the order will be the same across starts of the application, so configuring apps by monitor index cannot work reliably. Integrating such an API in winit would be a false promise that this would work.

Reply:
Ahh, I see, well then it's better to drop the set_center_monitor() function and use the Option<MonitorId> directly in the display config?